packages/mobile/src/components/CurrencyDisplay.tsx

```diff
@@ -131,9 +133,14 @@ export default function CurrencyDisplay({
   formatType,
   hideFullCurrencyName,
   style,
+  currencyInfo,
 }: Props) {
-  const localCurrencyCode = useLocalCurrencyCode()
-  const dollarToLocalRate = useDollarToLocalRate()
+  let localCurrencyCode = useLocalCurrencyCode()
+  let dollarToLocalRate = useDollarToLocalRate()
+  if (currencyInfo) {
+    localCurrencyCode = currencyInfo.currencyCode
```

Comment:
Could you name the currencyInfo something that makes it clear that it's transaction-specific? Maybe transactionCurrencyCode?